### PR TITLE
Configure Automated Plugin Release for OpenShift Plugins

### PR DESCRIPTION
**Pull request by coreydaley**
https://github.com/jenkinsci/openshift-sync-plugin
https://github.com/jenkinsci/openshift-client-plugin
https://github.com/jenkinsci/openshift-login-plugin

Commits configuring Automated Plugin Release in our OpenShift repositories:
https://github.com/jenkinsci/openshift-sync-plugin/commit/384d00db053f5e710c133dce53e1058bd43248cb
https://github.com/jenkinsci/openshift-client-plugin/commit/76e890f661cfeedc46e1d004c0e15d6fd051e336
https://github.com/jenkinsci/openshift-login-plugin/commit/1990221a79bbecb7e4c4f55f3591e14449199790
# Description

# Submitter checklist for adding or changing permissions

### Always

- [x] Add link to plugin/component Git repository in description above

### When enabling automated releases (cd: true)

- [x] Add a link to the pull request, which enables continuous delivery for your plugin or component.  
Follow the [documentation](https://www.jenkins.io/doc/developer/publishing/releasing-cd/) to ensure, your pull request is set up properly.

### When adding new uploaders (this includes newly created permissions files)

- [ ] [Make sure to `@`mention an existing maintainer to confirm the permissions request, if applicable](https://github.com/jenkins-infra/repository-permissions-updater/#requesting-permissions)
- [ ] Use the Jenkins community (LDAP) account name in the YAML file, not the GitHub account name
- [ ] Make sure to `@`mention the users being added so their GitHub account names are known if they require GitHub merge access (see below).
- [ ] [All newly added users have logged in to Artifactory and Jira at least once](https://github.com/jenkins-infra/repository-permissions-updater/#requesting-permissions)

### Reviewer checklist (not for requesters!)

- [ ] Check this if newly added person also needs to be given merge permission to the GitHub repo (please @ the people/person with their GitHub username in this issue as well). If needed, it can be done using an [IRC Bot command](https://jenkins.io/projects/infrastructure/ircbot/#github-repo-management)
- [ ] Check that the `$pluginId Developers` team has `Admin` permissions while granting the access.
- [ ] In the case of plugin adoption, ensure that the Jenkins Jira default assignee is either removed or changed to the new maintainer.
- [ ] If security contacts are changed (this includes add/remove), ping the security officer (currently `@Wadeck`) in this pull request. If an email contact is changed, wait for approval from the security officer.

There are [IRC Bot commands](https://jenkins.io/projects/infrastructure/ircbot/#issue-tracker-management) for it

### CODE REVIEW

File: permissions/plugin-openshift-login.yml [hunk 1]
```diff
@@ -10,3 +10,5 @@ developers:
 security:
   contacts:
     jira: "coreydaley"
+cd:
+  enabled: true"
```
Comment (NotMyFault) [inline permissions/plugin-openshift-login.yml:14]:
```suggestion
  enabled: true
```

**Comment by NotMyFault** [comment]
The commits linked are almost complete. Given the plugins haven't been incremental-ified before, you need to create a `.mvn/extensions.xml` in each repository manually with the following content: https://github.com/jenkinsci/archetypes/blob/master/common-files/.mvn/extensions.xml

**Comment by coreydaley** [comment]
@NotMyFault On this page https://www.jenkins.io/doc/developer/publishing/releasing-cd/ it mentions the .mvn/extensions.xml file, but doesn't tell you what the contents should be, as it does with the .mvn/maven.config file. I am sure with enough digging through the links and documentation it could be found, but I think it would be worth adding a section like there is for the .mvn/maven.config that just says "put these contents in this file". Just a suggestions.

**Comment by NotMyFault** [comment]
> but doesn't tell you what the contents should be, as it does with the .mvn/maven.config file

Commonly, you don't create these files manually. They are created if you incrementalify the plugin via `mvn incrementals:incrementalify`.
There's a link to the incrementals page in the same paragraph.

**Comment by coreydaley** [comment]
@NotMyFault I have added the .mvn/extensions.xml to all three repos.

File: permissions/plugin-openshift-login.yml (hunk 1 above)
Comment (coreydaley) [reply permissions/plugin-openshift-login.yml]: Ha, thanks for catching that!

**Review by NotMyFault — APPROVED** [review]

**Comment by coreydaley** [comment]
@NotMyFault About how long does it take for these updates to be deployed?

**Comment by NotMyFault** [comment]
> @NotMyFault About how long does it take for these updates to be deployed?

Once the check on the master branch is green, CD has been enabled for your repository.

**Pull request merged**